servlet/src/test/java/io/grpc/servlet/UndertowInteropTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.servlet;
+
+import static io.undertow.servlet.Servlets.defaultContainer;
+import static io.undertow.servlet.Servlets.deployment;
+import static io.undertow.servlet.Servlets.servlet;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.ServerBuilder;
+import io.grpc.internal.AbstractManagedChannelImplBuilder;
+import io.grpc.internal.AbstractServerImplBuilder;
+import io.grpc.testing.integration.AbstractInteropTest;
+import io.undertow.Handlers;
+import io.undertow.Undertow;
+import io.undertow.UndertowOptions;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.handlers.PathHandler;
+import io.undertow.servlet.api.DeploymentInfo;
+import io.undertow.servlet.api.DeploymentManager;
+import io.undertow.servlet.api.InstanceFactory;
+import io.undertow.servlet.util.ImmediateInstanceHandle;
+import java.net.InetSocketAddress;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Interop test for Undertow server and Netty client.
+ */
+public class UndertowInteropTest extends AbstractInteropTest {
+  private static final String HOST = "localhost";
+  private static final String MYAPP = "/grpc.testing.TestService";
+  private int port;
+  private Undertow server;
+  private DeploymentManager manager;
+
+  @After
+  @Override
+  public void tearDown() {
+    super.tearDown();
+    if (server != null) {
+      server.stop();
+    }
+    if (manager != null) {
+      try {
+        manager.stop();
+      } catch (ServletException e) {
+        throw new AssertionError("failed to stop container", e);
+      }
+    }
+  }
+
+  @Override
+  protected AbstractServerImplBuilder<?> getServerBuilder() {
+    return new ServletServerBuilder().maxInboundMessageSize(AbstractInteropTest.MAX_MESSAGE_SIZE);
+  }
+
+  @Override
+  protected void startServer(ServerBuilder<?> builer) {
```

Comment:
protected void startServer(ServerBuilder<?> builder)

change builer to builder on this line and the next?